### PR TITLE
Adds Contour Operator to Deployment Docs Page

### PR DESCRIPTION
**Pull request by danehans**
Adds Contour Operator to Contour's deployment docs page.

Signed-off-by: Daneyon Hansen <daneyonhansen@gmail.com>

### CODE REVIEW

**Comment by youngnick** [comment]
Could you rebase please @danehans ? I've fixed some CI problems on `main`, that should clear the failures on here.

File: site/docs/main/deploy-options.md [hunk 1]
```diff
@@ -10,7 +10,7 @@ If you don't have a cluster with that capability see the [Running without a Kube
 ### Recommended installation details
 
 The recommended installation of Contour is Contour running in a Deployment and Envoy in a Daemonset with TLS securing the gRPC communication between them.
-The [`contour` example][2] will install this for you.
+The [`contour` example][2] or [Contour Operator][12] will install this for you.
```
Comment (youngnick) [inline site/docs/main/deploy-options.md:13]: I'd like to see "the experimental" or similar here until the operator is GA.

**Review by youngnick — CHANGES_REQUESTED** [review]
Just a small thing about making sure we don't over-promote the operator until it's done.

**Comment by danehans** [comment]
@youngnick commit `7c601c7` resolves https://github.com/projectcontour/contour/pull/3140#discussion_r530682748, PTAL.

**Review by youngnick — APPROVED** [review]

**Pull request merged**